docs/public-api-changes-NIO1-to-NIO2.md

```diff
@@ -0,0 +1,7 @@
+# Changes in the Public API from NIO 1 to NIO 2
+
+- removed all previously deprecated functions, types and modules.
```

Comment:
Do we need to have an explicit list of removals?

Reply:
@Lukasa I don't think we need this for stuff that we had previously deprecated. I don't think we need to have a good migration story for people that use deprecated API.